sql/catalyst/src/test/scala/org/apache/spark/sql/connector/InMemoryTableCatalog.scala

```diff
@@ -63,35 +65,42 @@ class TestTableCatalog extends TableCatalog with SupportsNamespaces {
       schema: StructType,
       partitions: Array[Transform],
       properties: util.Map[String, String]): Table = {
-
     if (tables.containsKey(ident)) {
       throw new TableAlreadyExistsException(ident)
     }
 
-    if (partitions.nonEmpty) {
```

Comment:
the sql/core version supports partitions, so the merged version should support partitions as well.